crates/store/re_chunk_store/tests/dataframe.rs

```diff
@@ -136,3 +139,113 @@ fn schema_for_query() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn schema_static_columns() -> anyhow::Result<()> {
+    re_log::setup_logging();
+
+    let mut store = ChunkStore::new(
+        re_log_types::StoreId::random(re_log_types::StoreKind::Recording),
+        ChunkStoreConfig::COMPACTION_DISABLED,
+    );
+
+    let any_values = AnyValues::default()
+        .with_field("yak", Arc::new(StringArray::from(vec!["yuk"])))
+        .with_field("foo", Arc::new(StringArray::from(vec!["bar"])))
+        .with_field("baz", Arc::new(UInt32Array::from(vec![42u32])));
+
+    let entity_path = EntityPath::from("test");
+
+    let chunk0 = Chunk::builder(entity_path.clone())
+        .with_serialized_batches(
+            RowId::new(),
+            TimePoint::default(),
+            any_values.as_serialized_batches(),
+        )
+        .build()?;
+
+    store.insert_chunk(&Arc::new(chunk0))?;
+
+    let frame1 = TimeInt::new_temporal(1);
+    let chunk1 = Chunk::builder(entity_path)
+        .with_archetype(
+            RowId::new(),
+            [build_frame_nr(frame1)],
+            &MyPoints::new(MyPoint::from_iter(0..1)),
+        )
+        .build()?;
+
+    store.insert_chunk(&Arc::new(chunk1))?;
+
+    // Both
+
+    let query = QueryExpression {
+        view_contents: None,
+        include_semantically_empty_columns: false,
+        include_indicator_columns: false,
+        include_tombstone_columns: false,
+        include_static_columns: StaticColumnSelection::Both,
+        filtered_index: None,
+        filtered_index_range: None,
+        filtered_index_values: None,
+        using_index_values: None,
+        filtered_is_not_null: None,
+        sparse_fill_strategy: re_chunk_store::SparseFillStrategy::None,
+        selection: None,
+    };
+
+    let ChunkColumnDescriptors { components, .. } = store.schema_for_query(&query);
+
+    assert_eq!(
+        components
+            .iter()
+            .map(|column| (column.component_descriptor(), column.is_static))
+            .collect::<Vec<_>>(),
+        vec![
+            (ComponentDescriptor::partial("baz"), true),
+            (ComponentDescriptor::partial("foo"), true),
+            (ComponentDescriptor::partial("yak"), true),
+            (MyPoints::descriptor_points(), false),
+        ]
+    );
```

Comment:
this looks fine as is, but I think insta would have been a good match here, especially if it allows testing more of the `ChunkColumnDescriptors` structure. It does require a sane `Debug` impl though.